### PR TITLE
Session bug (fix with context variables)

### PR DESCRIPTION
**Pull request by nqn**

### CODE REVIEW

File: log10/load.py [hunk 1]
```diff
@@ -107,49 +108,53 @@ def get_session_id():
...
 
 
-# Global variable to store the current sessionID.
-sessionID = get_session_id()
-last_completion_response = None
-global_tags = []
+#
+# Context variables
+#
+session_id_var = contextvars.ContextVar("session_id", default=get_session_id())
```
Comment (michaeljin) [inline log10/load.py:114]: Slight preference to not set a default here and force an explicit set.

File: tests/test_context_managers.py [hunk 2]
```diff
@@ -0,0 +1,91 @@
+import asyncio
+import contextvars
+import uuid
+
+
+session_id_var = contextvars.ContextVar("session_id", default=str(uuid.uuid4()))
```
Comment (michaeljin) [inline tests/test_context_managers.py:6]: Same here - slight preference to not set a default. Makes it more explicit and simpler to test.

File: tests/test_context_managers.py [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+        session_id_var.reset(self.token)
+
+    async def __aenter__(self):
+        self.token = session_id_var.set(str(uuid.uuid4()))
+
+    async def __aexit__(self, exc_type, exc_value, exc_tb):
+        session_id_var.reset(self.token)
+
+
+def simulated_llm_call():
```
Comment (michaeljin) [inline tests/test_context_managers.py:27]: If you removed the default value - I'd expect to see an explicit set here (or in an outer scope) before get is possible.

**Review by michaeljin — APPROVED** [review]
Nice - haven't used contextvars before. Only change I'd be in favor of is to remove the default value for the contextvar. I think that's an easy way to get bugs that are difficult to reproduce.

File: log10/load.py (hunk 1 above)
Comment (nqn) [reply log10/load.py:114]: We need this to keep the existing behavior - where llm calls in outer most "context" i.e. no context, still gets grouped together

File: tests/test_context_managers.py (hunk 2 above)
Comment (nqn) [reply tests/test_context_managers.py:6]: This was just for testing, if that's ok to keep

File: tests/test_context_managers.py (hunk 3 above)
Comment (nqn) [reply tests/test_context_managers.py:27]: Note sure I follow. Is it to set uuids explicitly by having it as an input to the session object?

**Pull request merged**